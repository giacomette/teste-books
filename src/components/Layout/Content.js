import React from 'react';
import PropTypes from 'prop-types';
import { Container } from 'reactstrap';

function Content({ children }) {
  if (!children) return null;

  return (
    <div className="content-wrapper">
      <Container>{children}</Container>
    </div>
  );
}

Content.propTypes = {
  children: PropTypes.node
};

export default Content;
