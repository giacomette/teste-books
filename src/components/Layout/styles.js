import styled from 'styled-components';

export const Wrapper = styled.div`
  background: #f9f9f9;
  height: 100vh;

  * {
    font-family: 'Open Sans', 'Helvetica Neue', sans-serif;
  }
`;

export const ComtentWrapper = styled.div``;

export const WrapperHeader = styled.div`
  .navbar {
    background: #ea6b39;
    padding: 12px 0;
  }

  img {
    width: 120px;
    filter: brightness(500%) grayscale(10%);
  }

  .navbar-toggler-icon:before {
    content: "\f0c9";
    color: white;
  }
`;
