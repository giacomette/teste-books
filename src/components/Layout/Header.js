import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Container
} from 'reactstrap';
import logo from '../../assets/img/logo.svg';
import { WrapperHeader } from './styles';

function Header({ children }) {
  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => setIsOpen(!isOpen);

  return (
    <WrapperHeader>
      <Navbar expand="md">
        <Container>
          <NavbarBrand href="/">
            <img src={logo} alt="Supero" />
          </NavbarBrand>
          <NavbarToggler onClick={toggle} />
          {children ? (
            <HeaderContent isOpen={isOpen}>{children}</HeaderContent>
          ) : null}
        </Container>
      </Navbar>
    </WrapperHeader>
  );
}

function HeaderContent({ children, isOpen }) {
  if (!children) return null;

  return (
    <React.Fragment>
      <Collapse isOpen={isOpen} navbar>
        {children}
      </Collapse>
    </React.Fragment>
  );
}

Header.propTypes = {
  children: PropTypes.node
};

export default Header;
