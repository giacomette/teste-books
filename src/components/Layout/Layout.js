import React from 'react';
import PropTypes from 'prop-types';
import Header from './Header';
import Content from './Content';
import { Wrapper } from './styles';

function Layout({ header, content }) {
  return (
    <React.Fragment>
      <Wrapper>
        <Header>{header}</Header>
        <Content>{content}</Content>
      </Wrapper>
    </React.Fragment>
  );
}

Layout.propTypes = {
  header: PropTypes.node,
  content: PropTypes.node
};

export default Layout;
