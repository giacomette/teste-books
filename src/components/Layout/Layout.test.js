import React from 'react';
import Layout from './Layout';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('<Layout>', () => {
  it('deve renderizar sem haver alteração na estrutura do componente', () => {
    const component = renderer.create(<Layout />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('deve renderizar o conteudo do header caso exista', () => {
    const component = mount(<Layout header={<div>header</div>} />);

    const collapse = component.find('.navbar-collapse');

    expect(collapse.length).toBe(1);
  });

  it('não deve renderizar o conteudo do header caso não exista', () => {
    const component = mount(<Layout header={null} />);

    const collapse = component.find('.navbar-collapse');

    expect(collapse.length).toBe(0);
  });

  it('deve renderizar o conteudo do content do layout caso exista', () => {
    const component = mount(<Layout content={<div>content</div>} />);

    const content = component.find('.content-wrapper');

    expect(content.length).toBe(1);
  });

  it('não deve renderizar o conteudo do content do layout caso não exista', () => {
    const component = mount(<Layout content={null} />);

    const content = component.find('.content-wrapper');

    expect(content.length).toBe(0);
  });
});
