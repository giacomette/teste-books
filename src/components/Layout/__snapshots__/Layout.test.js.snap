// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout> deve renderizar sem haver alteração na estrutura do componente 1`] = `
<div
  className="sc-bdVaJa jeXhSc"
>
  <div
    className="sc-htpNat iIoqoj"
  >
    <nav
      className="navbar navbar-expand-md"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <img
            alt="Supero"
            src="logo.svg"
          />
        </a>
        <button
          aria-label="Toggle navigation"
          className="navbar-toggler"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
      </div>
    </nav>
  </div>
</div>
`;
